### PR TITLE
Add tests to checkstyle

### PR DESCRIPTION
**Pull request by tadgh**
- Add tests to checkstyle
- Suppress various hamcrest complaints until we fix them
- Name suppression modules:
  - HapiErrorCodeUniqueness
  - NoPrintln
- Work on suppressions:
  - unsuppress tests 
  - suppress hapi error checker in test files 
  - suppress hamcrest complaints in MDM 
- You can now suppress checkstyle warnings on classes/methods by annotating with the ID of the check.
```java
@SuppressWarnings("checkstyle:NoPrintln")
```

### CODE REVIEW

File: hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4ElasticTest.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public void testAutocompleteDirectionExisting() throws IOException {
 			ourLog.info("testAutocompleteDirectionExisting {}", text);
 			assertNotNull(valueSet);
 			List<ValueSet.ValueSetExpansionContainsComponent> expansions = valueSet.getExpansion().getContains();
-			// FIXME KHS
+			// TODO CHECKSTYLE KHS
 //			assertThat(expansions, hasItem(valueSetExpansionMatching(mean_blood_pressure)));
```
Comment (fil512) [inline hapi-fhir-jpaserver-elastic-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4ElasticTest.java:95]: ooh this one is important--leftover from the assertj work. i will need to get to that pronto

File: hapi-fhir-jpaserver-mdm/src/test/java/ca/uhn/fhir/jpa/mdm/svc/MdmGoldenResourceMergerSvcTest.java [hunk 2]
```diff
@@ -253,7 +253,7 @@ public void fromNoLinkToLink() {
 		Patient mergedSourcePatient = mergeGoldenPatients();
 		List<MdmLink> links = getNonRedirectLinksByGoldenResource(mergedSourcePatient);
 		assertThat(links).hasSize(1);
-		// FIXME KHS restore
+		// TODO CHECKSTYLE KHS restore
 //		assertThat(mergedSourcePatient, possibleLinkedTo(myTargetPatient1));
```
Comment (fil512) [inline hapi-fhir-jpaserver-mdm/src/test/java/ca/uhn/fhir/jpa/mdm/svc/MdmGoldenResourceMergerSvcTest.java:257]: can you do the mdm ones when you're doing that other hapi mdm stuff?

File: hapi-fhir-structures-dstu2/src/test/java/ca/uhn/fhir/util/FhirTerserDstu2Test.java [hunk 3]
```diff
@@ -793,7 +793,7 @@ public void testGetValuesWithTheCreateAndNoOverwrite() {
 		p.addUndeclaredExtension(true, "http://acme.org/modifierExtension", new StringDt("modifierValue"));
 		p.addUndeclaredExtension(false, "http://acme.org/parentExtension").addUndeclaredExtension(false, "http://acme.org/childExtension", new StringDt("nestedValue"));
 
-		System.out.println(ourCtx.newJsonParser().setPrettyPrint(true).encodeResourceToString(p));
+		ourLog.info(ourCtx.newJsonParser().setPrettyPrint(true).encodeResourceToString(p));
```
Comment (fil512) [inline hapi-fhir-structures-dstu2/src/test/java/ca/uhn/fhir/util/FhirTerserDstu2Test.java:796]: wow so many

File: test-job-template.yml [hunk 4]
```diff
@@ -27,7 +27,7 @@ jobs:
             inputs:
                goals: 'install'
                # These are Maven CLI options (and show up in the build logs) - "-nsu"=Don't update snapshots. We can remove this when Maven OSS is more healthy
-               options: '-P CI,FASTINSTALL -Dmaven.test.skip -e -B -Dmaven.repo.local=$(MAVEN_CACHE_FOLDER) -Dmaven.javadoc.skip=true -Dmaven.wagon.http.pool=false -Dhttp.keepAlive=false -Dstyle.color=always -Djansi.force=true'
+               options: '-P CI,CHECKSTYLE -Dmaven.test.skip -e -B -Dmaven.repo.local=$(MAVEN_CACHE_FOLDER) -Dmaven.javadoc.skip=true -Dmaven.wagon.http.pool=false -Dhttp.keepAlive=false -Dstyle.color=always -Djansi.force=true'
```
Comment (fil512) [inline test-job-template.yml:30]: so CHECKSTYLE profile now subsumes FASTINSTALL?

**Review by fil512 — APPROVED** [review]

File: hapi-fhir-jpaserver-mdm/src/test/java/ca/uhn/fhir/jpa/mdm/svc/MdmGoldenResourceMergerSvcTest.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-jpaserver-mdm/src/test/java/ca/uhn/fhir/jpa/mdm/svc/MdmGoldenResourceMergerSvcTest.java:257]: Yup

File: test-job-template.yml (hunk 4 above)
Comment (tadgh) [reply test-job-template.yml:30]: no, FASTINSTALL does nothing on hapi-fhir except for skip tests so /shrug.

Comment (tadgh) [reply test-job-template.yml:30]: image

File: hapi-fhir-jaxrsserver-base/src/test/java/ca/uhn/fhir/jaxrs/server/AbstractJaxRsConformanceProviderR4Test.java [hunk 5]
```diff
@@ -68,8 +70,8 @@ public void testConformanceWithMethods() throws Exception {
 		assertEquals(Constants.STATUS_HTTP_200_OK, response.getStatus());
 		assertThat(response.getEntity().toString()).contains("\"type\": \"Patient\"");
 		assertThat(response.getEntity().toString()).contains("\"someCustomOperation");
-		System.out.println(response);
-		System.out.println(response.getEntity());
```
[the remark this answers is not in the document]
Comment (tadgh) [reply hapi-fhir-jaxrsserver-base/src/test/java/ca/uhn/fhir/jaxrs/server/AbstractJaxRsConformanceProviderR4Test.java:72]: agreed!

File: test-job-template.yml (hunk 4 above)
Comment (tadgh) [reply test-job-template.yml:30]: actually, fastinstall specifically skips checkstyle entirely haha

File: hapi-fhir-checkstyle/pom.xml [hunk 6]
```diff
@@ -102,7 +102,7 @@
 			</build>
 		</profile>
 		<profile>
-			<id>CI</id>
+			<id>CHECKSTYLE</id>
```
[the remark this answers is not in the document]
Comment (tadgh) [reply hapi-fhir-checkstyle/pom.xml:105]: Agreed, I will add that to my list of tasks

**Pull request merged**